libs/designsystem/src/lib/components/popover/popover.component.ts

```diff
@@ -0,0 +1,210 @@
+import {
+  AfterViewInit,
+  Component,
+  ElementRef,
+  EventEmitter,
+  HostListener,
+  Inject,
+  Input,
+  OnChanges,
+  OnDestroy,
+  Output,
+  Renderer2,
+  SimpleChanges,
+  ViewChild,
+} from '@angular/core';
+
+import { DesignTokenHelper } from '../../helpers';
+
+export enum HorizontalDirection {
+  right = 'right',
+  left = 'left',
+}
+@Component({
+  selector: 'kirby-popover',
+  template: `
+    <div #wrapper class="wrapper"><ng-content></ng-content></div>
+  `,
+  styleUrls: ['./popover.component.scss'],
+})
+export class PopoverComponent implements AfterViewInit, OnChanges, OnDestroy {
+  @ViewChild('wrapper', { static: true, read: ElementRef })
+  wrapperElement: ElementRef<HTMLDivElement>;
+
+  @Input()
+  target: HTMLElement | ElementRef<HTMLElement>;
+
+  @Output()
+  willHide = new EventEmitter<void>();
+
+  private targetElement: HTMLElement;
+
+  private isShowing = false;
+
+  @Input()
+  popout: HorizontalDirection = HorizontalDirection.right;
+
+  @HostListener('click')
+  _backdropClick() {
+    this.willHide.emit();
+    this.hide();
+  }
+
+  @HostListener('window:resize')
+  _onWindowResize() {
+    if (this.isShowing) {
+      this.willHide.emit();
+      this.hide();
+    }
+  }
+
+  constructor(
+    private elementRef: ElementRef<HTMLElement>,
+    @Inject(DOCUMENT) private document: Document,
+    private renderer: Renderer2
+  ) {}
+
+  private zIndex = parseInt(DesignTokenHelper.zLayer('popover'));
+
+  ngAfterViewInit(): void {
+    const ionModal = this.elementRef.nativeElement.closest('ion-modal');
+    if (ionModal) {
+      this.zIndex = parseInt(ionModal.style.zIndex) + 1;
+      this.elementRef.nativeElement.style.zIndex = `${this.zIndex}`;
+    }
+    this.renderer.removeChild(
+      this.elementRef.nativeElement.parentElement,
+      this.elementRef.nativeElement
+    );
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.target.currentValue) {
+      this.targetElement =
+        this.target instanceof ElementRef ? this.target.nativeElement : this.target;
+    }
+  }
+
+  ngOnDestroy(): void {
+    this.hide();
+  }
+
+  // document.removeEventListener needs the exact same event handler & options reference:
+  private readonly preventScrollEventListenerOptions = {
+    passive: false,
+  } as EventListenerOptions;
+
+  // document.removeEventListener needs the exact same event handler & options reference:
+  private preventEvent(event: TouchEvent) {
+    event.preventDefault();
+  }
+
+  private preventScroll() {
+    this.renderer.addClass(this.document.body, 'backdrop-no-scroll'); // TODO use our own class instead of relying on Ionic?
+    this.document.addEventListener(
+      'touchmove',
+      this.preventEvent,
+      this.preventScrollEventListenerOptions
+    );
+  }
+
+  private releaseScroll() {
+    this.renderer.removeClass(this.document.body, 'backdrop-no-scroll'); // TODO use our own class instead of relying on Ionic?
```

Comment:
Consider TODO...

Reply:
How does this jive with no scroll on modal? Will it also release the scroll for that one?

Reply:
Do we want to lock down scrolling of the modal whenever the dropdown is placed inside it?

Reply:
Conclusion: it does not jive with the modal. 
We need to check for whether we're actually the first to apply the class.

Reply:
Or perhaps just a util function for starters to make it as simple as possible.